src/ratelimitedfunc.js

```diff
@@ -20,54 +20,28 @@ limitations under the License.
  * to update the interface once for all of them.
  *
  * Note that the function must not take arguments, since the args
- * could be different for each invocarion of the function.
+ * could be different for each invocation of the function.
  *
  * The returned function has a 'cancelPendingCall' property which can be called
  * on unmount or similar to cancel any pending update.
  */
-module.exports = function(f, minIntervalMs) {
-    this.lastCall = 0;
-    this.scheduledCall = undefined;
 
-    const self = this;
-    const wrapper = function() {
-        const now = Date.now();
-
-        if (self.lastCall < now - minIntervalMs) {
-            f.apply(this);
-            // get the time again now the function has finished, so if it
-            // took longer than the delay time to execute, it doesn't
-            // immediately become eligible to run again.
-            self.lastCall = Date.now();
-        } else if (self.scheduledCall === undefined) {
-            self.scheduledCall = setTimeout(
-                () => {
-                    self.scheduledCall = undefined;
-                    f.apply(this);
-                    // get time again as per above
-                    self.lastCall = Date.now();
-                },
-                (self.lastCall + minIntervalMs) - now,
-            );
-        }
-    };
-
-    // add the cancelPendingCall property
-    wrapper.cancelPendingCall = function() {
-        if (self.scheduledCall) {
-            clearTimeout(self.scheduledCall);
-            self.scheduledCall = undefined;
-        }
+import { throttle } from "lodash";
```

Comment:
oh, it's throttle here - is there a reason to use different ones in different places?

Reply:
Yeah, the behaviour we had in place here was the same as throttle, so I preserved that.
For a search field, debounce is usually better as you run the search code less and you can have a smaller delay.